File: docs/breez-sdk/snippets/flutter/lib/buying_bitcoin.dart
import 'package:breez_sdk_spark_flutter/breez_sdk_spark.dart';

Future<void> buyBitcoin(BreezSdk sdk) async {
  // ANCHOR: buy-bitcoin
  // Optionally, lock the purchase to a specific amount
  final optionalLockedAmountSat = BigInt.from(100000);
  // Optionally, set a redirect URL for after the purchase is completed
  final optionalRedirectUrl = "https://example.com/purchase-complete";

  final request = BuyBitcoinRequest_Moonpay(
      lockedAmountSat: optionalLockedAmountSat,
      redirectUrl: optionalRedirectUrl);

  final response = await sdk.buyBitcoin(request: request);
  print("Open this URL in a browser to complete the purchase:");
  print(response.url);
  // ANCHOR_END: buy-bitcoin
}

Future<void> buyBitcoinViaCashapp(BreezSdk sdk) async {
  // ANCHOR: buy-bitcoin-cashapp
  // Cash App requires the amount to be specified up front.
  final amountSats = BigInt.from(50000);

  final request = BuyBitcoinRequest_CashApp(amountSats: amountSats);

  final response = await sdk.buyBitcoin(request: request);
  print("Open this URL in Cash App to complete the purchase:");
  print(response.url);
  // ANCHOR_END: buy-bitcoin-cashapp
}

File: docs/breez-sdk/snippets/wasm/buying_bitcoin.ts
import {
  type BreezSdk
} from '@breeztech/breez-sdk-spark'

const buyBitcoin = async (sdk: BreezSdk) => {
  // ANCHOR: buy-bitcoin
  // Optionally, lock the purchase to a specific amount
  const optionalLockedAmountSat = 100_000
  // Optionally, set a redirect URL for after the purchase is completed
  const optionalRedirectUrl = 'https://example.com/purchase-complete'

  const response = await sdk.buyBitcoin({
    type: 'moonpay',
    lockedAmountSat: optionalLockedAmountSat,
    redirectUrl: optionalRedirectUrl
  })
  console.log('Open this URL in a browser to complete the purchase:')
  console.log(response.url)
  // ANCHOR_END: buy-bitcoin
}

const buyBitcoinViaCashapp = async (sdk: BreezSdk) => {
  // ANCHOR: buy-bitcoin-cashapp
  // Cash App requires the amount to be specified up front.
  const amountSats = 50_000

  const response = await sdk.buyBitcoin({
    type: 'cashApp',
    amountSats
  })
  console.log('Open this URL in Cash App to complete the purchase:')
  console.log(response.url)
  // ANCHOR_END: buy-bitcoin-cashapp
}
